crates/ide/src/references.rs

```diff
@@ -93,6 +93,8 @@ pub(crate) fn find_all_refs(
         get_struct_def_name_for_struct_literal_search(&sema, &syntax, position)
     {
         (Some(name), ReferenceKind::StructLiteral)
+    } else if let Some(res) = try_find_self_references(&syntax, position) {
+        return Some(res);
```

Comment:
The control flow in this one is weird. Let's just move this as a guarded return before the let?

Reply:
r=me with this fixed.